addons/stock/static/src/views/stock_orderpoint_list_controller.js

```diff
@@ -34,4 +34,15 @@ export class StockOrderpointListController extends ListController {
             onClose: () => { this.actionService.doAction({type: 'ir.actions.client', tag: 'reload'}); },
         });
     }
+
+    async createRecord() {
+        const domain = this.props.domain;
+        const location_id = domain.find((d) => d[0] === "location_id" && d[1] === "child_of")?.[2];
+        const list = this.model.root;
+
+        await super.createRecord();
+        if (list.editedRecord && location_id) {
+            list.editedRecord.update({ location_id: { id: location_id } });
```

Comment:
```suggestion
        if (this.editedRecord && location_id) {
            this.editedRecord.update({ location_id: { id: location_id } });
```
`editedRecord` is already available on `this`, no need for an `extra variable list`.

Reply:
Thanks for the review!

I have updated the code to use `this.editedRecord` directly. Also, I’ve included type checking for `location_id` to ensure safe updates and prevent potential errors.